Naming fixes

athipp opened the pull request:
Renaming the resource type based on PM recommendation.

athipp committed:
renaming resources and operations

amarzavery commented on arm-asazure/2016-05-06/swagger/asazure.json:
```diff
@@ -74,14 +74,7 @@
-          },
-          {
-            "name": "analysisServerName",
+            "name": "serverName",
```
looking at the sequence, I have usually seen resourcegroupName as the first parameter, then the RP specific artifact name, any query parameters and then the body parameter. If would be nice if you can follow the same sequence.
This sequence translates to the method signature.

athipp committed:
correcting version and parameter order

- Corrected the api-version number to be  2016-05-16,
- Corrected the ordering of paramters for CRUD operations, to match with
other specs.

tbombach commented:
Merged

AutorestCI commented:
No modification for NodeJS

AutorestCI commented:
No modification for Python

AutorestCI commented:
No modification for Ruby
